crates/rune-modules/src/time.rs

```diff
@@ -29,53 +29,671 @@
 //! }
 //! ```
 
-use rune::{Any, ContextError, Module};
+use rune::{docstring, runtime::Mut, Any, ContextError, Module};
+
+const NANOS_PER_SEC: u32 = 1_000_000_000;
 
 /// Construct the `time` module.
 pub fn module(_stdio: bool) -> Result<Module, ContextError> {
     let mut module = Module::with_crate("time")?;
-    module.ty::<Duration>()?;
-    module.function_meta(Duration::from_secs__meta)?;
-    module.function_meta(sleep)?;
+
+    module_duration(&mut module)?;
+    module_internal(&mut module)?;
+    module_instant(&mut module)?;
+
+    module.function_meta(sleep)?.docs(docstring! {
```

Comment:
You don't have to convert everything into using `docstring!`, since `function_meta` already picks it up if it's part of the function definition. It was only the associated constants (like `"SECOND"`) that could be improved.

Moving the docstrings here would prevent them from being visible to Rust users, so please move them back to the function definition.

Reply:
Okay, I assumed that `rune::function` would handle something like that, but I decided to switch everything to docstrings. I'll move it to a `Rust` instead.